www/js/control/ProfileSettings.jsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import { angularize } from "../angular-react-helper";
+import { object } from "prop-types";
+import ExpansionSection from "./ExpandMenu";
+import SettingRow from "./SettingRow";
+import ControlDataTable from "./ControlDataTable";
+import DemographicsSettingRow from "./DemographicsSettingRow";
+
+//any pure functions can go outside
+const ProfileSettings = ({ settingsScope, settingsObject }) => {
+    // anything that mutates must go in --- depend on props or state... 
+
+    // settingsScope is the $scope of general-settings.js
+    // grab any variables or functions you need from it like this:
+
+
+    //why is settings not defined but everything else is fine?
```

Comment:
I don't know why you would have had this problem, but it looks like you found a workaround?

I think that's fine for now. This way of accessing the $scope variables is temporary anyway.

Reply:
Settings is still undefined, even with the workaround that I was trying. I think it might be a timing issue, since settings is undefined in the moment ProfileSettings needs it, but the functions are fine. I've started moving as much as I can from general-settings into ProfileSettings, since the pass-through is temporary anyways. I'm thinking that will eventually resolve the issues I'm having here when everything is in one place.